lib/src/test/java/com/dynatrace/file/util/FilePollerTest.java

```diff
@@ -27,62 +71,111 @@ void filePollerConstructorThrowsOnPassingAFolder() throws IOException {
 
     assertThrows(
         IllegalArgumentException.class,
-        () -> {
-          new FilePoller(tempDir.toString());
-        });
+        () -> FilePollerFactory.getPollBased(tempDir.toString(), Duration.ofMillis(50)));
+
+    assertThrows(
+        IllegalArgumentException.class,
+        () -> FilePollerFactory.getWatchServiceBased(tempDir.toString()));
+  }
+
+  @Test
+  void filePollerUpdatesOnChangePollBased() throws IOException {
+    try (PollBasedFilePoller poller =
+        FilePollerFactory.getPollBased(tf.tempFile1Name(), Duration.ofMillis(50))) {
+      filePollerUpdatesOnChange(poller, tf.tempFile1());
+    }
   }
 
   @Test
-  void filePollerUpdatesOnChange() throws IOException, InterruptedException {
-    final Path tempFile = Files.createTempFile("tempfile", ".txt");
+  void filePollerUpdatesOnChangeWatchServiceBased() throws IOException {
+    if (IS_MAC_OS) {
+      System.out.println(
+          "macOS does not support WatchService based pollers, skipping this test...");
+      return;
+    }
```

Comment:
I just realized that we could use assumptions for this.

```suggestion
    assumeFalse(IS_MAC_OS, "macOS does not support WatchService based pollers, skipping this test...");
```
This will mark the test as "ignored" rather than "passed". Exit code will still be `0` in that case. :)

Reply:
Thats cool, I wasn't aware of that. Fixed, thanks!